lib/bot.js

```diff
@@ -129,9 +129,11 @@ class Bot {
       channels.forEach((channelName) => {
         const channel = channelName.toLowerCase();
         if (this.channelUsers[channel]) {
-          this.channelUsers[channel].delete(oldNick);
-          this.channelUsers[channel].add(newNick);
-          this.sendExactToDiscord(channel, `*${oldNick}* is now known as ${newNick}`);
+          if (this.channelUsers[channel].has(oldNick)) {
+            this.channelUsers[channel].delete(oldNick);
+            this.channelUsers[channel].add(newNick);
+            this.sendExactToDiscord(channel, `*${oldNick}* is now known as ${newNick}`);
+          }
         } else {
           logger.warn(`No channelUsers found for ${channel} when ${oldNick} changed.`);
```

Comment:
To be honest I don't think we really need the logger warning here, so maybe just fold the two `if`s into one and do `if (this.channelUsers[channel] && this.channelUsers[channel].has(oldNick)) {`?